### PR TITLE
fix some code

### PR DESCRIPTION
**Pull request by hunterhug**

### CODE REVIEW

File: redislock.go [hunk 1]
```diff
@@ -57,10 +56,14 @@ func (c *Client) Obtain(key string, ttl time.Duration, opt *Options) (*Lock, err
 		return nil, err
 	}
 
+	// Metadata string is appended to the lock token.
 	value := token + opt.getMetadata()
 	ctx := opt.getContext()
 
-	var backoff *time.Timer
+	// timer can be init first then defer stop
+	backoff := time.NewTimer(opt.getRetryBackoff())
```
Comment (dim) [inline redislock.go]: I don't want to allocate a timer if there is no need for it, hence the timer is allocated below on the first retry attempt. it's not as easy to read, but it's more memory-efficient

File: redislock.go [hunk 2]
```diff
@@ -85,8 +84,15 @@ func (c *Client) Obtain(key string, ttl time.Duration, opt *Options) (*Lock, err
 	return nil, ErrNotObtained
 }
 
+// alia method
+func (c *Client) Lock(key string, ttl time.Duration, opt *Options) (*Lock, error) {
```
Comment (dim) [inline redislock.go]: I don't like aliases, there's not reason for it

Comment (hunterhug) [reply redislock.go]: > I don't like aliases, there's not reason for it

The README.md example is better fix too, It has Lock().

Thank you.

Comment (dim) [reply redislock.go]: Thanks, noted.

**Pull request closed without merging**